src/Raven.Server/Dashboard/DatabasesInfoNotificationSender.cs

```diff
@@ -211,6 +222,37 @@ public static IEnumerable<AbstractDashboardNotification> FetchDatabasesInfo(Serv
                         SetOfflineDatabaseInfo(serverStore, transactionContext, databaseName, databasesInfo, drivesUsage, disabled: false);
                     }
                 }
+                
+                // 2. Add <system> info 
+                if (isValidFor == null) 
+                {
+                    var currentSystemHash = serverStore._env.CurrentReadTransactionId;
+                    if (currentSystemHash != CachedSystemInfo.Hash || CachedSystemInfo.NextDiskSpaceCheck <  SystemTime.UtcNow)
+                    {
+                        // Update cached value
+                        CachedSystemInfo.Hash = currentSystemHash;
+                        CachedSystemInfo.NextDiskSpaceCheck = SystemTime.UtcNow.AddSeconds(30);
+                        CachedSystemInfo.MountPoints.Clear();
```

Comment:
what happens if there is a concurrent read of this at the same time?

Reply:
At a minimum, you need to replace the whole object in one go

Reply:
Code refactored - See latest commit